File: examples/agents/simple_research_agent.ts
import { TWMarketDataClient } from "../../packages/js-sdk/src";

type ResearchContext = {
  symbol: string;
  issuerProfile: unknown;
  latestPriceRows: unknown;
  monthlyRevenueRows: unknown;
  trace: {
    issuerProfileRequestId: string | null;
    twsePriceRequestId: string | null;
    monthlyRevenueRequestId: string | null;
    dryRun: boolean | null;
    estimatedCreditsCost: number;
  };
};

async function run() {
  const apiKey = process.env.TWMD_API_KEY?.trim();
  if (!apiKey) {
    throw new Error("TWMD_API_KEY is required.");
  }

  const client = new TWMarketDataClient({
    apiKey,
    baseUrl: process.env.TWMD_BASE_URL || "https://twmarketdata.com",
    timeoutMs: 10000,
  });

  const symbol = process.env.TWMD_SYMBOL || "2330";

  const issuer = await client.issuerProfile({ symbol });
  const prices = await client.twseDailyPrice({ symbol, limit: 5 });
  const revenue = await client.monthlyRevenue({ symbol, limit: 12 });

  const context: ResearchContext = {
    symbol,
    issuerProfile: issuer.data,
    latestPriceRows: prices.data,
    monthlyRevenueRows: revenue.data,
    trace: {
      issuerProfileRequestId: issuer.meta.requestId,
      twsePriceRequestId: prices.meta.requestId,
      monthlyRevenueRequestId: revenue.meta.requestId,
      dryRun: issuer.meta.dryRun,
      estimatedCreditsCost: [issuer.meta.creditsCost, prices.meta.creditsCost, revenue.meta.creditsCost]
        .map((value) => value || 0)
        .reduce((sum, value) => sum + value, 0),
    },
  };

  console.log(JSON.stringify(context, null, 2));
}

run().catch((error) => {
  console.error("simple_research_agent_failed", error instanceof Error ? error.message : String(error));
  process.exit(1);
});

File: examples/agents/simple_research_agent.py
import json
import os

from twmarketdata import TWMarketDataClient


def main() -> None:
    api_key = (os.getenv("TWMD_API_KEY") or "").strip()
    if not api_key:
        raise RuntimeError("TWMD_API_KEY is required")

    symbol = os.getenv("TWMD_SYMBOL", "2330")
    client = TWMarketDataClient(
        api_key=api_key,
        base_url=os.getenv("TWMD_BASE_URL", "https://twmarketdata.com"),
        timeout=10,
    )

    issuer = client.issuer_profile(symbol=symbol)
    prices = client.twse_daily_price(symbol=symbol, limit=5)
    revenue = client.monthly_revenue(symbol=symbol, limit=12)

    context = {
        "symbol": symbol,
        "issuerProfile": issuer.data,
        "latestPriceRows": prices.data,
        "monthlyRevenueRows": revenue.data,
        "trace": {
            "issuerProfileRequestId": issuer.meta.request_id,
            "twsePriceRequestId": prices.meta.request_id,
            "monthlyRevenueRequestId": revenue.meta.request_id,
            "dryRun": issuer.meta.dry_run,
            "estimatedCreditsCost": (issuer.meta.credits_cost or 0)
            + (prices.meta.credits_cost or 0)
            + (revenue.meta.credits_cost or 0),
        },
    }

    print(json.dumps(context, ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()
